install/helm/agones/templates/service/allocation.yaml

```diff
@@ -276,3 +282,27 @@ data:
 {{- else }}
   tls-ca.crt: {{ .Files.Get "certs/allocator/server.crt" | b64enc }}
 {{- end }}
+
+---
+# Default allocation client secret
+{{- range .Values.gameservers.namespaces }}
+apiVersion: v1
+kind: Secret
+type: kubernetes.io/tls
+metadata:
+  name: allocator-client.default
+  namespace: {{ . }}
+  labels:
+    app: {{ template "agones.fullname" $ }}
+    chart: "{{ $.Chart.Name }}-{{ $.Chart.Version }}"
+    release: "{{ $.Release.Name }}"
+    heritage: "{{ $.Release.Service }}"
+data:
+{{- if $.Values.agones.allocator.generateClientTLS }}
+  tls.crt: {{ b64enc $selfSigned.Cert }}
+  tls.key: {{ b64enc $selfSigned.Key }}
+{{- else }}
+  tls.crt: {{ $.Files.Get "certs/allocator/allocator-client.default/client.crt" | b64enc }}
```

Comment:
I have a security concern that if someone uses install.yaml instead of helm chart to install Agones for production workload, their cluster is exposed as the secrets are publicly available. Sorry, I should have caught that earlier.

How about not generating the certificate if generateClientTLS is not set to `true`? In that case is there any way to test the secret in the e2e?

Reply:
Could we leave a warning here? https://github.com/googleforgames/agones/blob/master/install/helm/agones/templates/NOTES.txt

Reply:
I don't expect every user would read documentation.

Reply:
Okay, then no. There would be no way to test this in the e2e script.

Reply:
As in maybe, we could add logic here: https://github.com/googleforgames/agones/blob/master/build/Makefile#L283 such that lines beginning with tls.crt, tls.key, allocator-client.default.crt, ..., could be whitelisted from the diff.